src/main/java/org/assertj/core/api/AbstractLongAdderAssert.java

```diff
@@ -0,0 +1,331 @@
+
+import java.util.Comparator;
+import java.util.concurrent.atomic.LongAdder;
+
+import static org.assertj.core.error.ShouldHaveValue.shouldHaveValue;
+import static org.assertj.core.error.ShouldNotContainValue.shouldNotContainValue;
+
+/**
+ * Base class for all implementations of assertions for {@link LongAdder}s.
+ *
+ * @param <SELF> the "self" type of this assertion class. Please read &quot;<a href="http://bit.ly/1IZIRcY"
+ *               target="_blank">Emulating 'self types' using Java Generics to simplify fluent API implementation</a>&quot;
+ *               for more details.
+ *
+ * @author Grzegorz Piwowarek
+ */
+class AbstractLongAdderAssert<SELF extends AbstractLongAdderAssert<SELF>> extends AbstractAssert<SELF, LongAdder> implements NumberAssert<SELF, Long> {
+
+  @VisibleForTesting
+  Longs longs = Longs.instance();
+
+  public AbstractLongAdderAssert(LongAdder longAdder, Class<?> selfType) {
+    super(longAdder, selfType);
+  }
+
+  /**
+   * Verifies that the actual sum has a value strictly less than the given one.
+   * <p>
+   * Example:
+   * <pre><code class='java'>
+   * // assertions will pass:
+   * LongAdder actual1 = new LongAdder();
+   * actual1.add(1);
+   * assertThat(actual1).hasValueLessThan(2);
+   *
+   * LongAdder actual2 = new LongAdder();
+   * actual2.add(-2);
+   * assertThat(actual2).hasValueLessThan(-1);
+   *
+   * // assertions will fail:
+   * assertThat(actual1).hasValueLessThan(0)
+   *                    .hasValueLessThan(1);</code></pre>
+   *
+   * @param other the given value to compare the actual value to.
+   *
+   * @return {@code this} assertion object.
+   *
+   * @throws AssertionError if the actual adder is {@code null}.
+   * @throws AssertionError if the actual sum value is equal to or greater than the given one.
+   * @since 2.16.0 / 3.16.0
+   */
+  public SELF hasValueLessThan(long other) {
+    isNotNull();
+    longs.assertLessThan(info, actual.sum(), other);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual sum has a value strictly less than the given one.
+   * <p>
+   * Example:
+   * <pre><code class='java'>
+   * // assertions will pass:
+   * LongAdder actual1 = new LongAdder();
+   * actual1.add(1);
+   * assertThat(actual1).hasValueLessThanOrEqualTo(1)
+   *                    .hasValueLessThanOrEqualTo(2);
+   *
+   * LongAdder actual2 = new LongAdder();
+   * actual2.add(-2);
+   * assertThat(actual2).hasValueLessThanOrEqualTo(-1);
+   *
+   * // assertion will fail:
+   * assertThat(actual1).hasValueLessThanOrEqualTo(0);</code></pre>
+   *
+   * @param other the given value to compare the actual value to.
+   *
+   * @return {@code this} assertion object.
+   *
+   * @throws AssertionError if the actual adder is {@code null}.
+   * @throws AssertionError if the actual sum value is greater than the given one.
+   * @since 2.16.0 / 3.16.0
+   */
+  public SELF hasValueLessThanOrEqualTo(long other) {
+    isNotNull();
```

Comment:
I think `isNotNull()` can be avoided as `longs.assertLessThanOrEqualTo()` calls `Objects.instance().assertNotNull()`.

Same for the other methods.